### PR TITLE
fix(gh): make git-tag less error-prone by checking if the tag exist before pushing

### PR DESCRIPTION
**Pull request by LouisBrunner**
## Description

Fix issue in HC around release

## Review guidelines

Estimated Time of Review: 1 minute

### CODE REVIEW

File: .github/actions/git-tag/action.yml [hunk 1]
```diff
@@ -23,8 +23,15 @@ runs:
     - uses: actions/checkout@v3
       if: steps.checkout-test.outputs.checkout == 'true'
 
-    - name: Create tag
+    - name: Create tag it doesn't exist
```
Comment (domire8) [inline .github/actions/git-tag/action.yml:26]:
```suggestion
    - name: Create tag if it doesn't exist
```

File: .github/actions/git-tag/action.yml [hunk 2]
```diff
@@ -23,8 +23,15 @@ runs:
...
       run: |
+        if git ls-remote --exit-code origin refs/tags/${{ inputs.tag }}; then
+          echo "Tag already exists, skipping"
```
Comment (domire8) [inline .github/actions/git-tag/action.yml:29]:
```suggestion
          echo "::warning::Tag already exists, skipping"
```

File: .github/actions/git-tag/action.yml [hunk 3]
```diff
@@ -23,8 +23,15 @@ runs:
...
+          exit 0
+        fi
+
+        echo "Tag doesn't exist, creating"
```
Comment (domire8) [inline .github/actions/git-tag/action.yml:33]:
```suggestion
        echo "::debug::Tag doesn't exist, creating"
```

**Review by domire8 — APPROVED** [review]
Looks good

File: .github/actions/git-tag/action.yml (hunk 3 above)
Comment (LouisBrunner) [reply .github/actions/git-tag/action.yml:33]: Not sure if this should be debug, having to rerun the workflow in a different mode is kind of the opposite of what we want for this?

Comment (LouisBrunner) [reply .github/actions/git-tag/action.yml:33]: I just leave them as-is personally, it's good for logs you can check when something goes wrong but you don't need a whole annotation from GitHub (which is a good point for a skip warning above)

**Pull request merged**